integration-tests/src/test/java/com/datastax/oss/driver/api/core/metadata/SerializableMetadataIT.java

```diff
@@ -0,0 +1,152 @@
+            "CREATE TABLE smaller (val int primary key, data numbers_data)",
+            "CREATE INDEX doub_idx on all (doub)",
+            "CREATE OR REPLACE FUNCTION avgState ( state tuple<int,bigint>, val int ) CALLED ON NULL INPUT RETURNS tuple<int,bigint> LANGUAGE java AS"
+                + " 'if (val !=null) { state.setInt(0, state.getInt(0)+1); state.setLong(1, state.getLong(1)+val.intValue()); } return state;'",
+            "CREATE OR REPLACE FUNCTION avgFinal ( state tuple<int,bigint> ) CALLED ON NULL INPUT RETURNS double LANGUAGE java AS"
+                + " 'double r = 0; if (state.getInt(0) == 0) return null; r = state.getLong(1); r/= state.getInt(0); return Double.valueOf(r);'",
+            "CREATE AGGREGATE IF NOT EXISTS average ( int ) SFUNC avgState STYPE tuple<int,bigint> FINALFUNC avgFinal INITCOND (0,0)",
+            "INSERT INTO all (val,doub,str) VALUES (1,2,'one')",
+            "INSERT INTO all (val,doub,str) VALUES (2,4,'two')",
+            "INSERT INTO all (val,doub,str) VALUES (3,6,'three')",
+            "CREATE MATERIALIZED VIEW IF NOT EXISTS big AS SELECT * FROM all"
+                + " WHERE val is not null and doub is not null and val > 2"
+                + " PRIMARY KEY (val,doub)");
+
+    CqlSession session = sessionRule.session();
+    statements.forEach(session::execute);
+  }
+
+  private KeyspaceMetadata getAndValidateKsMetadata() {
+
+    Optional<KeyspaceMetadata> ksOption =
+            sessionRule.session().getMetadata().getKeyspace(sessionRule.session().getKeyspace().get());
+    assertTrue(ksOption.isPresent());
+    KeyspaceMetadata ks = ksOption.get();
+    assertTrue(ks instanceof DefaultKeyspaceMetadata);
+    return ks;
+  }
+
+  @Test
+  public void should_contain_serializable_metadata() {
+
+    KeyspaceMetadata ks = getAndValidateKsMetadata();
+
+    /* Validate that the keyspace metadata is fully populated */
+    assertFalse(ks.getUserDefinedTypes().isEmpty());
+    assertFalse(ks.getTables().isEmpty());
+    assertFalse(ks.getViews().isEmpty());
+    assertFalse(ks.getFunctions().isEmpty());
+    assertFalse(ks.getAggregates().isEmpty());
+
+    Optional<TableMetadata> tableOption = ks.getTable("all");
+    assertTrue(tableOption.isPresent());
+    TableMetadata table = tableOption.get();
+    assertTrue(table instanceof DefaultTableMetadata);
+
+    /* Validate that the table metadata is fully populated */
+    assertFalse(table.getPartitionKey().isEmpty());
+    assertFalse(table.getClusteringColumns().isEmpty());
+    assertFalse(table.getColumns().isEmpty());
+    // TODO: What kind of objects can be set for options here?  How do we guarantee they're
+    // serializable?
+    assertFalse(table.getOptions().isEmpty());
+    assertFalse(table.getIndexes().isEmpty());
+
+    SerializationUtils.roundtrip((DefaultKeyspaceMetadata) ks);
+  }
+
+  @Test
+  public void should_support_describe_for_aggregates_with_deserialized_instances() {
```

Comment:
This test already passed due to the exception handling in DefaultAggregateMetadata.describe().  Adding this as an explicit integration test as a guard against future breakage in functionality (since that's the only area affected by the transient declaration).